cpp/include/cudf/aggregation.hpp

```diff
@@ -87,7 +87,8 @@ class aggregation {
     CUDA,            ///< CUDA UDF based reduction
     MERGE_LISTS,     ///< merge multiple lists values into one list
     MERGE_SETS,      ///< merge multiple lists values into one list then drop duplicate entries
-    MERGE_M2         ///< merge partial values of M2 aggregation
+    MERGE_M2,        ///< merge partial values of M2 aggregation,
+    CORR,            ///< correlation among multiple columns
```

Comment:
```suggestion
    CORRELATION,            ///< correlation between two sets of elements
```

Reply:
If it helps preemptively avoid a future breaking change, we might want to consider being more explicit in the API about Pearson vs other correlation aggregations (such as spearman).

Reply:
I had the same concern, but my quick googling didn't make it seem like there were other notions of correlation. Seems like there are, so we should name accordingly.

Reply:
updated. Added the `correlation_type` as an enum to `make_correlation_aggregation` factory.